### ISSUE TITLE
[Feature request] Proposal for support react-leaflet markers

### ISSUE THREAD

**Issue by carlosrj** [YUzhva/react-leaflet-markercluster#17]
This is a feature request - #16 PR

**Current behavior:**

Actually react-leaflet-markercluster component just accept to add markers as specific json object.

**Example code:**

```jsx
const marker = [
    {lat: 49.8397, lng: 24.0297}
];
<MarkerClusterGroup
    markers={marker}
/>
```

**Expected behavior:**

_Thinking in component development and react architecture_, in my personal opinion, I'd insert markers or any component that can be processed through component itself, in this case, [react-leaflet marker](https://github.com/PaulLeCam/react-leaflet/blob/master/docs/Components.md#marker).

**Example code:**

```jsx
const marker = [
    {lat: 49.8397, lng: 24.0297}
];
<MarkerClusterGroup>
    <Marker position={[marker.lat, marke.lng]} />
</MarkerClusterGroup>
```

I just created a PR with the proposal development, I hope that can help.
Thank you for your consideration.

**Comment by YUzhva** [comment]
@carlosrj Thank you, for contribution.

This is the really nice suggestion, also was thinking about the opportunity to add Markers not only through .json object.

I need some time for reviewing and publishing that feature...

**Comment by carlosrj** [comment]
@YUzhva thanks for your time.

About react-leaflet already has Marker component, it's in example marker component, i.e., it's already integrated in PR :smile:

**Comment by FelipeKohlmann** [comment]
I was doing some tests with this new feature and I found that when the state of changes in my component, the position of a marker for example, the MarkerClusterGroup doesn't render again with the updates, but if I remove the MarkerClusterGroup tag the markers render and update as intended.

**Comment by carlosrj** [comment]
@FelipeKohlmann thanks for report.

I'll check this issue, although I'm sure that will appear more. I was doing tests with other react-leaflet plugins and there are some bugs. If you can contribute with a possible solution, it will be great.

I've no much time these weeks, but surely I'll check it.

**Comment by YUzhva** [comment]
The other way of clustering layers is to access not to the marker through the reference of react component (too heavy way) but accessing to the leafletElement of the marker that was passed as a child.

Check this, react-leaflet library creating leaflet marker before rendering:
https://github.com/PaulLeCam/react-leaflet/blob/master/src/Marker.js#L30

And that object, that was created by marker function from leaflet lib. is exactly what we need for clustering (that is the object that we need send to L.markerClusterGroup(LEAFLET_MARKER_OBJECT))

The only issue, that I can't find that opportunity on react-leaflet library, so I created next issue:
https://github.com/PaulLeCam/react-leaflet/issues/309

**Comment by YUzhva** [comment]
@carlosrj 

Implemented as testing feature in: [release] 1.1.4: react-leaflet Marker support (#18)

**Comment by carlosrj** [comment]
@YUzhva great job!

I just tested it with some react-leaflet components as children and it's working absolutely good :ok_hand:
